### Import packages

In [22]:
import pandas as pd
import numpy as np

### Load responses

In [31]:
df = pd.read_csv('irony_annotation_study_responses.csv')

In [32]:
# df = df[:2]

df

,Timestamp,Wie alt bist du?,Wie oft nutzt du Reddit?,Wie gut schätzt du dich selbst ein Ironie im Alltag erkennen zu können?,Wie gut kennst du r/de?,KOMMENTAR #1\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Solange alle ihre Tiefkühlpizza und RTL haben ist doch alles super!\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?,KOMMENTAR #2\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Ich bin schockiert.\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.1,"KOMMENTAR #3\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Würden sich die babyboomer zu Tode arbeiten, bräuchte man ihnen keine Rente mehr zahlen\n\nWie ironisch ist dieser Kommentar?",...,KOMMENTAR #56\n\nTHREAD: EU-Einigung scheitert: Freiwillige Chatkontrolle endet\n\nKOMMENTAR: Für heute?\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.55,KOMMENTAR #57\n\nTHREAD: EU-Einigung scheitert: Freiwillige Chatkontrolle endet\n\nKOMMENTAR: Soll ich nochmal?\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.56,KOMMENTAR #58\n\nTHREAD: CDU fordert Aufklärung über Verwendung des Sondervermögens\n\nKOMMENTAR: Da kann Reiche gleich die nächste Arbeitsgruppe gründen um das ganze zu verschleppen\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.57,KOMMENTAR #59\n\nTHREAD: Verschwörungs-Demo in Berlin: Xavier Naidoo singt - deutlich weniger Teilnehmende als angekündigt\n\nKOMMENTAR: Gabs da auch embryonal gewürzte Currywurst mit Pommes?\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.58,KOMMENTAR #60\n\nTHREAD: Verschwörungs-Demo in Berlin: Xavier Naidoo singt - deutlich weniger Teilnehmende als angekündigt\n\nKOMMENTAR: Dieser Weg ist peinlich und leer.\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.59
0,18/03/2026 14:32:07,18-24,Selten,5,"Kenne ich, lese selten",5,5,3,3,5,...,5,4,5,4,4,4,5,4,5,4
1,18/03/2026 14:33:51,18-24,"Kenne Reddit, nutze es kaum",4,Kenne ich nicht,4,4,3,3,3,...,5,5,5,5,4,2,5,5,2,5
2,18/03/2026 14:35:06,18-24,Selten,5,"Kenne ich, lese selten",4,4,3,4,2,...,4,5,5,5,4,4,5,5,1,3
3,18/03/2026 14:35:12,18-24,"Kenne Reddit, nutze es kaum",4,Kenne ich nicht,4,4,3,4,2,...,4,4,4,5,3,3,5,5,5,5
4,18/03/2026 14:35:38,35-44,"Kenne Reddit, nutze es kaum",4,Kenne ich nicht,5,4,4,3,5,...,3,2,2,1,4,3,3,4,3,3
5,18/03/2026 14:36:38,18-24,"Kenne Reddit, nutze es kaum",3,Kenne ich nicht,5,4,1,4,3,...,5,4,5,4,3,3,5,5,5,5
6,18/03/2026 14:37:08,18-24,Kenne Reddit nicht,3,Kenne ich nicht,4,4,1,4,3,...,5,5,5,5,3,3,5,5,4,2
7,18/03/2026 14:39:48,25-34,Täglich,5,"Kenne ich, lese selten",5,4,4,4,5,...,5,5,3,2,4,5,5,4,5,5
8,18/03/2026 14:39:48,25-34,"Kenne Reddit, nutze es kaum",4,Kenne ich nicht,5,5,3,3,5,...,4,4,4,3,4,3,5,5,2,4


### Split data into general info and comment rating

In [33]:
general_info = df.iloc[:, :4]

general_info

,Timestamp,Wie alt bist du?,Wie oft nutzt du Reddit?,Wie gut schätzt du dich selbst ein Ironie im Alltag erkennen zu können?
0,18/03/2026 14:32:07,18-24,Selten,5
1,18/03/2026 14:33:51,18-24,"Kenne Reddit, nutze es kaum",4
2,18/03/2026 14:35:06,18-24,Selten,5
3,18/03/2026 14:35:12,18-24,"Kenne Reddit, nutze es kaum",4
4,18/03/2026 14:35:38,35-44,"Kenne Reddit, nutze es kaum",4
5,18/03/2026 14:36:38,18-24,"Kenne Reddit, nutze es kaum",3
6,18/03/2026 14:37:08,18-24,Kenne Reddit nicht,3
7,18/03/2026 14:39:48,25-34,Täglich,5
8,18/03/2026 14:39:48,25-34,"Kenne Reddit, nutze es kaum",4


In [34]:
comment_rating = df.iloc[:, 5:]

comment_rating

,KOMMENTAR #1\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Solange alle ihre Tiefkühlpizza und RTL haben ist doch alles super!\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?,KOMMENTAR #2\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Ich bin schockiert.\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.1,"KOMMENTAR #3\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Würden sich die babyboomer zu Tode arbeiten, bräuchte man ihnen keine Rente mehr zahlen\n\nWie ironisch ist dieser Kommentar?",Wie sicher bist du dir bei dieser Einschätzung?.2,KOMMENTAR #4\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Auf den Schreck erstmal eine Rentenerhöhung.\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.3,KOMMENTAR #5\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Finde den Generationen-Vertrag an sich schon dämlich. Will meine eigene Rente auffüllen und nicht die von anderen.\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.4,...,KOMMENTAR #56\n\nTHREAD: EU-Einigung scheitert: Freiwillige Chatkontrolle endet\n\nKOMMENTAR: Für heute?\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.55,KOMMENTAR #57\n\nTHREAD: EU-Einigung scheitert: Freiwillige Chatkontrolle endet\n\nKOMMENTAR: Soll ich nochmal?\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.56,KOMMENTAR #58\n\nTHREAD: CDU fordert Aufklärung über Verwendung des Sondervermögens\n\nKOMMENTAR: Da kann Reiche gleich die nächste Arbeitsgruppe gründen um das ganze zu verschleppen\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.57,KOMMENTAR #59\n\nTHREAD: Verschwörungs-Demo in Berlin: Xavier Naidoo singt - deutlich weniger Teilnehmende als angekündigt\n\nKOMMENTAR: Gabs da auch embryonal gewürzte Currywurst mit Pommes?\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.58,KOMMENTAR #60\n\nTHREAD: Verschwörungs-Demo in Berlin: Xavier Naidoo singt - deutlich weniger Teilnehmende als angekündigt\n\nKOMMENTAR: Dieser Weg ist peinlich und leer.\n\nWie ironisch ist dieser Kommentar?,Wie sicher bist du dir bei dieser Einschätzung?.59
0,5,5,3,3,5,4,5,5,2,3,...,5,4,5,4,4,4,5,4,5,4
1,4,4,3,3,3,3,5,5,1,1,...,5,5,5,5,4,2,5,5,2,5
2,4,4,3,4,2,3,5,5,1,4,...,4,5,5,5,4,4,5,5,1,3
3,4,4,3,4,2,4,4,3,1,4,...,4,4,4,5,3,3,5,5,5,5
4,5,4,4,3,5,5,5,5,2,3,...,3,2,2,1,4,3,3,4,3,3
5,5,4,1,4,3,2,5,4,1,5,...,5,4,5,4,3,3,5,5,5,5
6,4,4,1,4,3,3,5,5,1,4,...,5,5,5,5,3,3,5,5,4,2
7,5,4,4,4,5,3,4,3,1,4,...,5,5,3,2,4,5,5,4,5,5
8,5,5,3,3,5,4,2,1,1,5,...,4,4,4,3,4,3,5,5,2,4


In [35]:
# Python indexing: start:stop:step
irony = comment_rating.iloc[:, ::2] # every 2nd column starting at 0
confidence = comment_rating.iloc[:, 1::2] # every 2nd column starting at 1

In [36]:
irony

,KOMMENTAR #1\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Solange alle ihre Tiefkühlpizza und RTL haben ist doch alles super!\n\nWie ironisch ist dieser Kommentar?,KOMMENTAR #2\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Ich bin schockiert.\n\nWie ironisch ist dieser Kommentar?,"KOMMENTAR #3\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Würden sich die babyboomer zu Tode arbeiten, bräuchte man ihnen keine Rente mehr zahlen\n\nWie ironisch ist dieser Kommentar?",KOMMENTAR #4\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Auf den Schreck erstmal eine Rentenerhöhung.\n\nWie ironisch ist dieser Kommentar?,KOMMENTAR #5\n\nTHREAD: Mehrheit sieht ältere im Vorteil - nicht einmal ein Drittel der jungen glaubt an faire Rente\n\nKOMMENTAR: Finde den Generationen-Vertrag an sich schon dämlich. Will meine eigene Rente auffüllen und nicht die von anderen.\n\nWie ironisch ist dieser Kommentar?,KOMMENTAR #6\n\nTHREAD: Solarbranche: Reiche-Pläne gefährden Zehntausende Jobs\n\nKOMMENTAR: ist ja nicht so als wären es jobs in der automobilindustrie also warum die aufregung\n\nWie ironisch ist dieser Kommentar?,KOMMENTAR #7\n\nTHREAD: Solarbranche: Reiche-Pläne gefährden Zehntausende Jobs\n\nKOMMENTAR: Leute alles gut. Laut der EU Präsidentin kriegen wir bald eh alle kleine Mini AKWs und dann jeder seinen eigenen Atompilz im Keller\n\nWie ironisch ist dieser Kommentar?,KOMMENTAR #8\n\nTHREAD: Solarbranche: Reiche-Pläne gefährden Zehntausende Jobs\n\nKOMMENTAR: Die Öl&Gas Lobby macht halt n guten Job und die Reiche ist dankbares Opfer!\n\nWie ironisch ist dieser Kommentar?,KOMMENTAR #9\n\nTHREAD: Solarbranche: Reiche-Pläne gefährden Zehntausende Jobs\n\nKOMMENTAR: Vergeht denn kein Tag wo dieser Roboter keine Dummheit macht\n\nWie ironisch ist dieser Kommentar?,"KOMMENTAR #10\n\nTHREAD: Ölkonzerne klagen gegen Untersuchung durch Bundeskartellamt\n\nKOMMENTAR: Wer Probleme mit niedrigem Blutdruck hat, sollte den Artikel lesen. Dann pumpt es wieder ordentlich.\n\nWie ironisch ist dieser Kommentar?",...,"KOMMENTAR #51\n\nTHREAD: Psychotherapie: Wer hier kürzt, produziert mehr Krankheit und Leid\n\nKOMMENTAR: Ja, aber ihr müsst das einfach durch die Wirtschaftskompetenzbrille der Union betrachten. Wenn man das jetzt kürzt, gibts auch weniger Diagnosen, die das Jobcenter anzweifeln muss.\n\nWie ironisch ist dieser Kommentar?","KOMMENTAR #52\n\nTHREAD: Psychotherapie: Wer hier kürzt, produziert mehr Krankheit und Leid\n\nKOMMENTAR: Sehen Sie demnächst in diesem Theater: Oh nein, wo kommen denn die ganzen Obdachlosen her?\n\nWie ironisch ist dieser Kommentar?","KOMMENTAR #53\n\nTHREAD: Wehrdienst: Deutlich mehr Menschen bewerben sich bei der Bundeswehr\n\nKOMMENTAR: Wenn der Arbeitsmarkt schwächelt, steigt die Attraktivität von sicheren Arbeitgebern. Ganz normale und nachvollziehbare Entwicklung eigentlich.\n\nWie ironisch ist dieser Kommentar?","KOMMENTAR #54\n\nTHREAD: Wehrdienst: Deutlich mehr Menschen bewerben sich bei der Bundeswehr\n\nKOMMENTAR: Also sollten wir schleunigst den Wehrdienst wieder einführen, weil Freiwilligkeit offensichtlich nicht funktioniert, wenn die Anreize stimmen. Zwang ist der beste Weg\n\nWie ironisch ist dieser Kommentar?",KOMMENTAR #55\n\nTHREAD: EU-Einigung scheitert: Freiwillige Chatkontrolle endet\n\nKOMMENTAR: Bis zum nächsten Versuch.\n\nWie ironisch ist dieser Kommentar?,KOMMENTAR #56\n\nTHREAD: EU-Einigung scheitert: Freiwillige Chatkontrolle endet\n\nKOMMENTAR: Für heute?\n\nWie ironisch ist dieser Kommentar?,KOMMENTAR #57\n\nTHREAD: EU-Einigung scheitert: Freiwillige Chatkontrolle endet\n\nKOMMENTAR: Soll ich nochmal?\n\nWie ironisch ist dieser Kommentar?,KOMMENTAR #58\n\nTHREAD: CDU fordert Aufklärung über Verwendung des Sonder

In [37]:
confidence

,Wie sicher bist du dir bei dieser Einschätzung?,Wie sicher bist du dir bei dieser Einschätzung?.1,Wie sicher bist du dir bei dieser Einschätzung?.2,Wie sicher bist du dir bei dieser Einschätzung?.3,Wie sicher bist du dir bei dieser Einschätzung?.4,Wie sicher bist du dir bei dieser Einschätzung?.5,Wie sicher bist du dir bei dieser Einschätzung?.6,Wie sicher bist du dir bei dieser Einschätzung?.7,Wie sicher bist du dir bei dieser Einschätzung?.8,Wie sicher bist du dir bei dieser Einschätzung?.9,...,Wie sicher bist du dir bei dieser Einschätzung?.50,Wie sicher bist du dir bei dieser Einschätzung?.51,Wie sicher bist du dir bei dieser Einschätzung?.52,Wie sicher bist du dir bei dieser Einschätzung?.53,Wie sicher bist du dir bei dieser Einschätzung?.54,Wie sicher bist du dir bei dieser Einschätzung?.55,Wie sicher bist du dir bei dieser Einschätzung?.56,Wie sicher bist du dir bei dieser Einschätzung?.57,Wie sicher bist du dir bei dieser Einschätzung?.58,Wie sicher bist du dir bei dieser Einschätzung?.59
0,5,3,4,5,3,3,5,4,4,5,...,4,4,4,4,4,4,4,4,4,4
1,4,3,3,5,1,2,5,2,4,4,...,3,4,5,5,5,5,5,2,5,5
2,4,4,3,5,4,4,4,4,4,4,...,4,4,5,3,4,5,5,4,5,3
3,4,4,4,3,4,4,5,3,3,5,...,5,5,3,4,4,4,5,3,5,5
4,4,3,5,5,3,4,5,2,1,3,...,3,3,5,3,1,2,1,3,4,3
5,4,4,2,4,5,3,5,3,1,4,...,3,5,3,5,3,4,4,3,5,5
6,4,4,3,5,4,3,3,3,3,3,...,3,4,4,4,5,5,5,3,5,2
7,4,4,3,3,4,2,5,3,3,4,...,3,4,2,3,4,5,2,5,4,5
8,5,3,4,1,5,3,5,4,1,4,...,5,3,4,4,3,4,3,3,5,4


## Measurements

### Cronbach's Alpha
#### Reliability coefficient and a measure of the internal consistency of our survey
Do our items measure the same construct (irony)?

In [38]:
def cronbach_alpha(df):
    k = df.shape[1]
    variance_sum = df.var(axis=0, ddof=1).sum()
    total_var = df.sum(axis=1).var(ddof=1)

    return (k / (k - 1)) * (1 - (variance_sum / total_var))

In [39]:
alpha_irony = cronbach_alpha(irony)
alpha_confidence = cronbach_alpha(confidence)

In [40]:
alpha_irony

np.float64(0.8082579570525372)

In [41]:
alpha_confidence

np.float64(0.8973080757726819)

- Irony alpha ≈ 0.81 --> good
  - Die Teilnehmer interpretieren Ironie über alle Kommentare hinweg auf einheitliche Weise
- Confidence ≈ 0.9 --> very good
  - Sehr hohe Konsistenz
  - Die Konfidenz verhält sich sehr stabil